### PR TITLE
Update contributor guidelines

### PR DESCRIPTION
**Pull request by khaeru**
The branch ``JamesFrierson1:master`` used in #202 was not set to allow maintainer push access, so this PR supersedes that one.

### CODE REVIEW

**Comment by khaeru** [comment]
@behnam2015 since you and @OFR-IIASA wrote the original draft of the tutorial guidelines, can you please check the revisions I've made to them here? Thank you!

File: CONTRIBUTING.rst [hunk 1]
```diff
@@ -1,35 +1,220 @@
...
+
+- Tutorials are formatted as Jupyter notebooks in Python or R.
+- Commit 'bare' notebooks in git, i.e. without cell output.
+  Notebooks will be run and rendered when the documentation is generated.
+- Add a line to ``tests/test_tutorials.py``, so that the parametrized test function runs the tutorial (as noted at `#196 <https://github.com/iiasa/message_ix/pull/196>`_).
+- Optionally, use Jupyter notebook slide-show features so that the tutorial can be viewed as a presentation.
+- When relevant, provide links to publications or sources that provide greater detail for the methodology, data, or other packages used.
+- Providing the mathematical formulation in the tutorial itself is optional.
+- Framework specific variables and parameters or functions must be in italic.
+
```
Comment (behnam2015) [inline CONTRIBUTING.rst]: @khaeru, if needed we can mention:
"- Adding relevant figures and tables in the tutorial is recoemmnded if that helps the users to understand the concept more easily." And perhaps, you can add a note mentioning where these sumpplementary figures/tables should be loaded as raw files.

File: CONTRIBUTING.rst [hunk 2]
```diff
@@ -1,35 +1,220 @@
...
+  - Reference and provide links to any tutorials that are interlinked or part of a series.
+
+- Description of individual steps:
+
+  - A brief explanation of the step.
+  - A link to any relevant mathematical documentation.
+
+- Results:
+
+  - Results should be retrieved using the generic reporting tool.
```
Comment (behnam2015) [inline CONTRIBUTING.rst]: @khaeru, maybe clarifying this further by adding a link to the reporting tool and/or mentioning "...using MESSAGEix generic reporting tool".

File: CONTRIBUTING.rst [hunk 3]
```diff
@@ -1,35 +1,220 @@
...
+  - Plots to depict results should use `pyam <https://github.com/IAMconsortium/pyam/>`_.
+
```
Comment (behnam2015) [inline CONTRIBUTING.rst]: @khaeru, I suggest adding:
" - It is recommended to briefly discuss some of the insights from the results in line with the learning objectives of the tutorial.
- Some follow-up exercises and questions can be added at the end of the tutorial (optional)."

**Comment by khaeru** [comment]
Thanks @behnam2015 for the suggestions, and @JamesFrierson1 for getting this started.

**Pull request merged**